### PR TITLE
docs: describe how the limiter is configured, and how limits are scaled

### PR DESCRIPTION
**Pull request by marten-seemann**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -281,64 +281,78 @@ aid understanding of applicable limits.  Note that the (wrapped) error
...
+scale with the size of the machine.
+
+This is done using the `ScalingLimitConfig`. For every scope, this configuration
+struct defines the absolutely bare minimum limits, and an (optional) increase of
+these limits, which will be applied on nodes that have sufficient memory.
+
+A `ScalingLimitConfig` can be converted into a `LimitConfig` (which can then be
+used to initialize a fixed limiter as shown above) by calling the `Scale` method.
+The `Scale` method takes two parameters: the amount of memory and the number of file
+descriptors.that an application is willing to dedicate to libp2p.
```
Comment (MarcoPolo) [inline README.md:305]:
```suggestion
descriptors that an application is willing to dedicate to libp2p.
```

File: README.md [hunk 2]
```diff
@@ -281,64 +281,78 @@ aid understanding of applicable limits.  Note that the (wrapped) error
...
+
+These amounts will differ between use cases: A blockchain node running on a dedicated
+server might have a lot of memory, and dedicate 1/4 of that memory to libp2p. On the
+other end of the spectrum, a desktop companion application running as a background
+task on a consumer laptop will probably dedicate significantly less than 1/4 of its system
+memory to libp2p.
+
+For convenience, the `ScalingLimitConfig` also provides an `AutoScale` method,
+which determines the amount of memory and file descriptors available on the
+system, and dedicates 1/8 of the memory and 1/2 of the file descriptors to libp2p.
```
Comment (MarcoPolo) [inline README.md:315]:
```suggestion
system, and dedicates up to 1/8 of the memory and 1/2 of the file descriptors to libp2p.
```

File: README.md [hunk 3]
```diff
@@ -281,64 +281,78 @@ aid understanding of applicable limits.  Note that the (wrapped) error
...
+  },
+}
 ```
 
-We also provide a dynamic limiter which uses the same base limits, but
-the memory limit is dynamically computed at each memory reservation check
-based on free memory.
+The base limit (`SystemBaseLimit`) here is the minimum configuration that any
+node will have, no matter how little memory it possesses. For every GB of memory
+passed into the `Scale` method, the increase is (`SystemLimitIncrease`) is added.
```
Comment (MarcoPolo) [inline README.md:345]:
```suggestion
passed into the `Scale` method, an increase  of (`SystemLimitIncrease`) is added.
```

File: README.md [hunk 4]
```diff
@@ -281,64 +281,78 @@ aid understanding of applicable limits.  Note that the (wrapped) error
...
+
+Example: Calling `Scale` with 4 GB of memory will result in a limit of 384 for
```
Comment (MarcoPolo) [inline README.md:347]:
```suggestion
For Example, calling `Scale` with 4 GB of memory will result in a limit of 384 for
```

**Review by MarcoPolo — APPROVED** [review]
lgtm mod minor grammar suggestions.

File: README.md [hunk 5]
```diff
@@ -281,64 +281,78 @@ aid understanding of applicable limits.  Note that the (wrapped) error
...
+The base limit (`SystemBaseLimit`) here is the minimum configuration that any
+node will have, no matter how little memory it possesses. For every GB of memory
+passed into the `Scale` method, an increase  of (`SystemLimitIncrease`) is added.
+
+For Example, calling `Scale` with 4 GB of memory will result in a limit of 384 for
+`Conns` (128 + 4*64).
+
+The `FDFraction` defines how many of the file descriptors are allocated to this
+scope. In the example above, when called with a file descriptor value of 1000,
+this would result in a limit of 1256 file descriptors for the system scope.
```
Comment (BigLep) [inline README.md:352]: How did we get to 1256?

**Review by BigLep — COMMENTED** [review]
This is helpful to see written out - thanks.  Other things I think I would be curious about as a user:
1. How to bypass limits
2. What are the default limits if I do nothing when constructing a ResourceManager (or even better  as a user am I always forced to set a  Limiter when constructing a ResourceManager thus need to think about this).
3. If I do always have to create a Limiter, is there a base LimitConfig I can use (and where is it)?
4. What does the code look like if I like everything about a base LimitConfig but then want to override one specific part of it.

**Pull request merged**